src/EventStore.Core/DataStructures/ProbabilisticFilter/AlignedMemory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace EventStore.Core.DataStructures.ProbabilisticFilter {
+	// net6 can do aligned allocations directly
+	// but we need to overallocate and do our own alignment.
+	public sealed unsafe class AlignedMemory : IDisposable {
```

Comment:
@timothycoleman does a finalizer need to be added as a precaution to free up the allocated memory in case someone forgets to call Dispose() ?

Reply:
Yeah you’re right, I’ll take a look, probable make use of a safe handle

Reply:
Actually I think let’s get this in as is if you’re happy and we can circle back round after the release